homepage/tests.py

```diff
@@ -103,3 +106,145 @@ def test_create_multiple_courses():
     # make sure that multiple courses are saved accurately
     courses = [Course.objects.get(pk=arg[0]).get_details() for arg in details]
     assert courses == details
+
+
+# ---------Professor tests----------#
+
+@pytest.fixture
+def professors_list():
+    return list(Professor.objects.values_list(
+        'name'))
+
+
+@pytest.fixture
+def valid_name():
+    return 'Severus Snape'
+
+
+@pytest.fixture
+def invalid_name():
+    return 'Pablo Diego Jose Francisco de Paula Juan Nepomuceno Maria de \
+    los Remedios Cipriano de la Santisima Trinidad Ruiz y Picasso Snape'
+
+
+# check the data stored in database from 0004_professor_test_data.py file
+@pytest.mark.django_db
+def test_model_professors_list(professors_list):
+
+    assert professors_list == [
+        ('Septima Vector',),
+        ('Sybill Patricia Trelawney',),
+        ('Bathsheda Babbling',),
+    ]
+
+
+@pytest.mark.django_db
+def test_add_valid_professor(valid_name):
+    professor = Professor(name=valid_name)
+    professor.save()

```

Comment:
if you're testing adding a valid name, maybe test that it doesn't raise an exception?

Reply:
An exception will fail the test automatically